// upstox-js-sdk/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function () {
    return _ApiClient.ApiClient;
  }
});
Object.defineProperty(exports, "ApiGatewayErrorResponse", {
  enumerable: true,
  get: function () {
    return _ApiGatewayErrorResponse.ApiGatewayErrorResponse;
  }
});
Object.defineProperty(exports, "BrokerageData", {
  enumerable: true,
  get: function () {
    return _BrokerageData.BrokerageData;
  }
});
Object.defineProperty(exports, "BrokerageTaxes", {
  enumerable: true,
  get: function () {
    return _BrokerageTaxes.BrokerageTaxes;
  }
});
Object.defineProperty(exports, "BrokerageWrapperData", {
  enumerable: true,
  get: function () {
    return _BrokerageWrapperData.BrokerageWrapperData;
  }
});
Object.defineProperty(exports, "CancelOrderData", {
  enumerable: true,
  get: function () {
    return _CancelOrderData.CancelOrderData;
  }
});
Object.defineProperty(exports, "CancelOrderResponse", {
  enumerable: true,
  get: function () {
    return _CancelOrderResponse.CancelOrderResponse;
  }
});
Object.defineProperty(exports, "ChargeApi", {
  enumerable: true,
  get: function () {
    return _ChargeApi.ChargeApi;
  }
});
Object.defineProperty(exports, "ConvertPositionData", {
  enumerable: true,
  get: function () {
    return _ConvertPositionData.ConvertPositionData;
  }
});
Object.defineProperty(exports, "ConvertPositionRequest", {
  enumerable: true,
  get: function () {
    return _ConvertPositionRequest.ConvertPositionRequest;
  }
});
Object.defineProperty(exports, "ConvertPositionResponse", {
  enumerable: true,
  get: function () {
    return _ConvertPositionResponse.ConvertPositionResponse;
  }
});
Object.defineProperty(exports, "Depth", {
  enumerable: true,
  get: function () {
    return _Depth.Depth;
  }
});
Object.defineProperty(exports, "DepthMap", {
  enumerable: true,
  get: function () {
    return _DepthMap.DepthMap;
  }
});
Object.defineProperty(exports, "DpPlan", {
  enumerable: true,
  get: function () {
    return _DpPlan.DpPlan;
  }
});
Object.defineProperty(exports, "GetBrokerageResponse", {
  enumerable: true,
  get: function () {
    return _GetBrokerageResponse.GetBrokerageResponse;
  }
});
Object.defineProperty(exports, "GetFullMarketQuoteResponse", {
  enumerable: true,
  get: function () {
    return _GetFullMarketQuoteResponse.GetFullMarketQuoteResponse;
  }
});
Object.defineProperty(exports, "GetHistoricalCandleResponse", {
  enumerable: true,
  get: function () {
    return _GetHistoricalCandleResponse.GetHistoricalCandleResponse;
  }
});
Object.defineProperty(exports, "GetHoldingsResponse", {
  enumerable: true,
  get: function () {
    return _GetHoldingsResponse.GetHoldingsResponse;
  }
});
Object.defineProperty(exports, "GetIntraDayCandleResponse", {
  enumerable: true,
  get: function () {
    return _GetIntraDayCandleResponse.GetIntraDayCandleResponse;
  }
});
Object.defineProperty(exports, "GetMarketQuoteLastTradedPriceResponse", {
  enumerable: true,
  get: function () {
    return _GetMarketQuoteLastTradedPriceResponse.GetMarketQuoteLastTradedPriceResponse;
  }
});
Object.defineProperty(exports, "GetMarketQuoteOHLCResponse", {
  enumerable: true,
  get: function () {
    return _GetMarketQuoteOHLCResponse.GetMarketQuoteOHLCResponse;
  }
});
Object.defineProperty(exports, "GetOrderBookResponse", {
  enumerable: true,
  get: function () {
    return _GetOrderBookResponse.GetOrderBookResponse;
  }
});
Object.defineProperty(exports, "GetOrderResponse", {
  enumerable: true,
  get: function () {
    return _GetOrderResponse.GetOrderResponse;
  }
});
Object.defineProperty(exports, "GetPositionResponse", {
  enumerable: true,
  get: function () {
    return _GetPositionResponse.GetPositionResponse;
  }
});
Object.defineProperty(exports, "GetProfileResponse", {
  enumerable: true,
  get: function () {
    return _GetProfileResponse.GetProfileResponse;
  }
});
Object.defineProperty(exports, "GetProfitAndLossChargesResponse", {
  enumerable: true,
  get: function () {
    return _GetProfitAndLossChargesResponse.GetProfitAndLossChargesResponse;
  }
});
Object.defineProperty(exports, "GetTradeResponse", {
  enumerable: true,
  get: function () {
    return _GetTradeResponse.GetTradeResponse;
  }
});
Object.defineProperty(exports, "GetTradeWiseProfitAndLossDataResponse", {
  enumerable: true,
  get: function () {
    return _GetTradeWiseProfitAndLossDataResponse.GetTradeWiseProfitAndLossDataResponse;
  }
});
Object.defineProperty(exports, "GetTradeWiseProfitAndLossMetaDataResponse", {
  enumerable: true,
  get: function () {
    return _GetTradeWiseProfitAndLossMetaDataResponse.GetTradeWiseProfitAndLossMetaDataResponse;
  }
});
Object.defineProperty(exports, "GetUserFundMarginResponse", {
  enumerable: true,
  get: function () {
    return _GetUserFundMarginResponse.GetUserFundMarginResponse;
  }
});
Object.defineProperty(exports, "HistoricalCandleData", {
  enumerable: true,
  get: function () {
    return _HistoricalCandleData.HistoricalCandleData;
  }
});
Object.defineProperty(exports, "HistoryApi", {
  enumerable: true,
  get: function () {
    return _HistoryApi.HistoryApi;
  }
});
Object.defineProperty(exports, "HoldingsData", {
  enumerable: true,
  get: function () {
    return _HoldingsData.HoldingsData;
  }
});
Object.defineProperty(exports, "IntraDayCandleData", {
  enumerable: true,
  get: function () {
    return _IntraDayCandleData.IntraDayCandleData;
  }
});
Object.defineProperty(exports, "LoginApi", {
  enumerable: true,
  get: function () {
    return _LoginApi.LoginApi;
  }
});
Object.defineProperty(exports, "LogoutResponse", {
  enumerable: true,
  get: function () {
    return _LogoutResponse.LogoutResponse;
  }
});
Object.defineProperty(exports, "MarketQuoteApi", {
  enumerable: true,
  get: function () {
    return _MarketQuoteApi.MarketQuoteApi;
  }
});
Object.defineProperty(exports, "MarketQuoteOHLC", {
  enumerable: true,
  get: function () {
    return _MarketQuoteOHLC.MarketQuoteOHLC;
  }
});
Object.defineProperty(exports, "MarketQuoteSymbol", {
  enumerable: true,
  get: function () {
    return _MarketQuoteSymbol.MarketQuoteSymbol;
  }
});
Object.defineProperty(exports, "MarketQuoteSymbolLtp", {
  enumerable: true,
  get: function () {
    return _MarketQuoteSymbolLtp.MarketQuoteSymbolLtp;
  }
});
Object.defineProperty(exports, "ModifyOrderData", {
  enumerable: true,
  get: function () {
    return _ModifyOrderData.ModifyOrderData;
  }
});
Object.defineProperty(exports, "ModifyOrderRequest", {
  enumerable: true,
  get: function () {
    return _ModifyOrderRequest.ModifyOrderRequest;
  }
});
Object.defineProperty(exports, "ModifyOrderResponse", {
  enumerable: true,
  get: function () {
    return _ModifyOrderResponse.ModifyOrderResponse;
  }
});
Object.defineProperty(exports, "OAuthClientException", {
  enumerable: true,
  get: function () {
    return _OAuthClientException.OAuthClientException;
  }
});
Object.defineProperty(exports, "OAuthClientExceptionCause", {
  enumerable: true,
  get: function () {
    return _OAuthClientExceptionCause.OAuthClientExceptionCause;
  }
});
Object.defineProperty(exports, "OAuthClientExceptionCauseStackTrace", {
  enumerable: true,
  get: function () {
    return _OAuthClientExceptionCauseStackTrace.OAuthClientExceptionCauseStackTrace;
  }
});
Object.defineProperty(exports, "OAuthClientExceptionCauseSuppressed", {
  enumerable: true,
  get: function () {
    return _OAuthClientExceptionCauseSuppressed.OAuthClientExceptionCauseSuppressed;
  }
});
Object.defineProperty(exports, "Ohlc", {
  enumerable: true,
  get: function () {
    return _Ohlc.Ohlc;
  }
});
Object.defineProperty(exports, "OrderApi", {
  enumerable: true,
  get: function () {
    return _OrderApi.OrderApi;
  }
});
Object.defineProperty(exports, "OrderBookData", {
  enumerable: true,
  get: function () {
    return _OrderBookData.OrderBookData;
  }
});
Object.defineProperty(exports, "OrderData", {
  enumerable: true,
  get: function () {
    return _OrderData.OrderData;
  }
});
Object.defineProperty(exports, "OtherTaxes", {
  enumerable: true,
  get: function () {
    return _OtherTaxes.OtherTaxes;
  }
});
Object.defineProperty(exports, "PlaceOrderData", {
  enumerable: true,
  get: function () {
    return _PlaceOrderData.PlaceOrderData;
  }
});
Object.defineProperty(exports, "PlaceOrderRequest", {
  enumerable: true,
  get: function () {
    return _PlaceOrderRequest.PlaceOrderRequest;
  }
});
Object.defineProperty(exports, "PlaceOrderResponse", {
  enumerable: true,
  get: function () {
    return _PlaceOrderResponse.PlaceOrderResponse;
  }
});
Object.defineProperty(exports, "PortfolioApi", {
  enumerable: true,
  get: function () {
    return _PortfolioApi.PortfolioApi;
  }
});
Object.defineProperty(exports, "PositionData", {
  enumerable: true,
  get: function () {
    return _PositionData.PositionData;
  }
});
Object.defineProperty(exports, "Problem", {
  enumerable: true,
  get: function () {
    return _Problem.Problem;
  }
});
Object.defineProperty(exports, "ProfileData", {
  enumerable: true,
  get: function () {
    return _ProfileData.ProfileData;
  }
});
Object.defineProperty(exports, "ProfitAndLossChargesData", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossChargesData.ProfitAndLossChargesData;
  }
});
Object.defineProperty(exports, "ProfitAndLossChargesTaxes", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossChargesTaxes.ProfitAndLossChargesTaxes;
  }
});
Object.defineProperty(exports, "ProfitAndLossChargesWrapperData", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossChargesWrapperData.ProfitAndLossChargesWrapperData;
  }
});
Object.defineProperty(exports, "ProfitAndLossMetaData", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossMetaData.ProfitAndLossMetaData;
  }
});
Object.defineProperty(exports, "ProfitAndLossMetaDataWrapper", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossMetaDataWrapper.ProfitAndLossMetaDataWrapper;
  }
});
Object.defineProperty(exports, "ProfitAndLossOtherChargesTaxes", {
  enumerable: true,
  get: function () {
    return _ProfitAndLossOtherChargesTaxes.ProfitAndLossOtherChargesTaxes;
  }
});
Object.defineProperty(exports, "TokenRequest", {
  enumerable: true,
  get: function () {
    return _TokenRequest.TokenRequest;
  }
});
Object.defineProperty(exports, "TokenResponse", {
  enumerable: true,
  get: function () {
    return _TokenResponse.TokenResponse;
  }
});
Object.defineProperty(exports, "TradeData", {
  enumerable: true,
  get: function () {
    return _TradeData.TradeData;
  }
});
Object.defineProperty(exports, "TradeProfitAndLossApi", {
  enumerable: true,
  get: function () {
    return _TradeProfitAndLossApi.TradeProfitAndLossApi;
  }
});
Object.defineProperty(exports, "TradeWiseMetaData", {
  enumerable: true,
  get: function () {
    return _TradeWiseMetaData.TradeWiseMetaData;
  }
});
Object.defineProperty(exports, "TradeWiseProfitAndLossData", {
  enumerable: true,
  get: function () {
    return _TradeWiseProfitAndLossData.TradeWiseProfitAndLossData;
  }
});
Object.defineProperty(exports, "UserApi", {
  enumerable: true,
  get: function () {
    return _UserApi.UserApi;
  }
});
Object.defineProperty(exports, "UserFundMarginData", {
  enumerable: true,
  get: function () {
    return _UserFundMarginData.UserFundMarginData;
  }
});
Object.defineProperty(exports, "WebsocketApi", {
  enumerable: true,
  get: function () {
    return _WebsocketApi.WebsocketApi;
  }
});
Object.defineProperty(exports, "WebsocketAuthRedirectResponse", {
  enumerable: true,
  get: function () {
    return _WebsocketAuthRedirectResponse.WebsocketAuthRedirectResponse;
  }
});
Object.defineProperty(exports, "WebsocketAuthRedirectResponseData", {
  enumerable: true,
  get: function () {
    return _WebsocketAuthRedirectResponseData.WebsocketAuthRedirectResponseData;
  }
});
var _ApiClient = require("./ApiClient");
var _ApiGatewayErrorResponse = require("./model/ApiGatewayErrorResponse");
var _BrokerageData = require("./model/BrokerageData");
var _BrokerageTaxes = require("./model/BrokerageTaxes");
var _BrokerageWrapperData = require("./model/BrokerageWrapperData");
var _CancelOrderData = require("./model/CancelOrderData");
var _CancelOrderResponse = require("./model/CancelOrderResponse");
var _ConvertPositionData = require("./model/ConvertPositionData");
var _ConvertPositionRequest = require("./model/ConvertPositionRequest");
var _ConvertPositionResponse = require("./model/ConvertPositionResponse");
var _Depth = require("./model/Depth");
var _DepthMap = require("./model/DepthMap");
var _DpPlan = require("./model/DpPlan");
var _GetBrokerageResponse = require("./model/GetBrokerageResponse");
var _GetFullMarketQuoteResponse = require("./model/GetFullMarketQuoteResponse");
var _GetHistoricalCandleResponse = require("./model/GetHistoricalCandleResponse");
var _GetHoldingsResponse = require("./model/GetHoldingsResponse");
var _GetIntraDayCandleResponse = require("./model/GetIntraDayCandleResponse");
var _GetMarketQuoteLastTradedPriceResponse = require("./model/GetMarketQuoteLastTradedPriceResponse");
var _GetMarketQuoteOHLCResponse = require("./model/GetMarketQuoteOHLCResponse");
var _GetOrderBookResponse = require("./model/GetOrderBookResponse");
var _GetOrderResponse = require("./model/GetOrderResponse");
var _GetPositionResponse = require("./model/GetPositionResponse");
var _GetProfileResponse = require("./model/GetProfileResponse");
var _GetProfitAndLossChargesResponse = require("./model/GetProfitAndLossChargesResponse");
var _GetTradeResponse = require("./model/GetTradeResponse");
var _GetTradeWiseProfitAndLossDataResponse = require("./model/GetTradeWiseProfitAndLossDataResponse");
var _GetTradeWiseProfitAndLossMetaDataResponse = require("./model/GetTradeWiseProfitAndLossMetaDataResponse");
var _GetUserFundMarginResponse = require("./model/GetUserFundMarginResponse");
var _HistoricalCandleData = require("./model/HistoricalCandleData");
var _HoldingsData = require("./model/HoldingsData");
var _IntraDayCandleData = require("./model/IntraDayCandleData");
var _LogoutResponse = require("./model/LogoutResponse");
var _MarketQuoteOHLC = require("./model/MarketQuoteOHLC");
var _MarketQuoteSymbol = require("./model/MarketQuoteSymbol");
var _MarketQuoteSymbolLtp = require("./model/MarketQuoteSymbolLtp");
var _ModifyOrderData = require("./model/ModifyOrderData");
var _ModifyOrderRequest = require("./model/ModifyOrderRequest");
var _ModifyOrderResponse = require("./model/ModifyOrderResponse");
var _OAuthClientException = require("./model/OAuthClientException");
var _OAuthClientExceptionCause = require("./model/OAuthClientExceptionCause");
var _OAuthClientExceptionCauseStackTrace = require("./model/OAuthClientExceptionCauseStackTrace");
var _OAuthClientExceptionCauseSuppressed = require("./model/OAuthClientExceptionCauseSuppressed");
var _Ohlc = require("./model/Ohlc");
var _OrderBookData = require("./model/OrderBookData");
var _OrderData = require("./model/OrderData");
var _OtherTaxes = require("./model/OtherTaxes");
var _PlaceOrderData = require("./model/PlaceOrderData");
var _PlaceOrderRequest = require("./model/PlaceOrderRequest");
var _PlaceOrderResponse = require("./model/PlaceOrderResponse");
var _PositionData = require("./model/PositionData");
var _Problem = require("./model/Problem");
var _ProfileData = require("./model/ProfileData");
var _ProfitAndLossChargesData = require("./model/ProfitAndLossChargesData");
var _ProfitAndLossChargesTaxes = require("./model/ProfitAndLossChargesTaxes");
var _ProfitAndLossChargesWrapperData = require("./model/ProfitAndLossChargesWrapperData");
var _ProfitAndLossMetaData = require("./model/ProfitAndLossMetaData");
var _ProfitAndLossMetaDataWrapper = require("./model/ProfitAndLossMetaDataWrapper");
var _ProfitAndLossOtherChargesTaxes = require("./model/ProfitAndLossOtherChargesTaxes");
var _TokenRequest = require("./model/TokenRequest");
var _TokenResponse = require("./model/TokenResponse");
var _TradeData = require("./model/TradeData");
var _TradeWiseMetaData = require("./model/TradeWiseMetaData");
var _TradeWiseProfitAndLossData = require("./model/TradeWiseProfitAndLossData");
var _UserFundMarginData = require("./model/UserFundMarginData");
var _WebsocketAuthRedirectResponse = require("./model/WebsocketAuthRedirectResponse");
var _WebsocketAuthRedirectResponseData = require("./model/WebsocketAuthRedirectResponseData");
var _ChargeApi = require("./api/ChargeApi");
var _HistoryApi = require("./api/HistoryApi");
var _LoginApi = require("./api/LoginApi");
var _MarketQuoteApi = require("./api/MarketQuoteApi");
var _OrderApi = require("./api/OrderApi");
var _PortfolioApi = require("./api/PortfolioApi");
var _TradeProfitAndLossApi = require("./api/TradeProfitAndLossApi");
var _UserApi = require("./api/UserApi");
var _WebsocketApi = require("./api/WebsocketApi");
